[media-kit] media-gfx/exiv2 Add app-arch/brotli dependency to exiv2 t…

org-tekeli-borisp opened the pull request:
…emplate

This change includes app-arch/brotli as a dependency in the exiv2 template for improved functionality or compatibility. The update ensures all required libraries are correctly referenced.

org-tekeli-borisp committed:
[media-kit] media-gfx/exiv2 Add app-arch/brotli dependency to exiv2 template

This change includes app-arch/brotli as a dependency in the exiv2 template for improved functionality or compatibility. The update ensures all required libraries are correctly referenced.

geaaru commented:
Could be this PR incomplete? I don't see autogen.yml.
Can you also remove reference to `mark/packages.yaml` and `packages.yaml` and curated files?

org-tekeli-borisp committed:
[media-kit] media-gfx/exiv2 Add autogen.yaml for media-gfx/exiv2 in media-kit

Introduces an autogen.yaml for automating exiv2 package generation. This utilizes metadata from GitHub to streamline package updates. Simplifies maintenance by reducing manual intervention.

org-tekeli-borisp commented:
> Could be this PR incomplete? I don't see autogen.yml. Can you also remove reference to `mark/packages.yaml` and `packages.yaml` and curated files?

Indeed!

org-tekeli-borisp commented:
added autogen.yaml

geaaru commented:
I think that we can remove *test* IUSE and `src_test`

org-tekeli-borisp committed:
[media-kit] media-gfx/exiv2 Remove test support from exiv2 template.

This commit removes the test-related dependencies, restrictions, and the `src_test` function from the exiv2 template. This simplifies the template by excluding the testing functionality, which is no longer required.

org-tekeli-borisp commented:
done

geaaru commented on media-kit/mark/media-gfx/exiv2/templates/exiv2.tmpl:
```diff
@@ -0,0 +1,98 @@
+
+SLOT="0/$(ver_cut 1-2)"
+KEYWORDS="*"
+IUSE="+bmff doc examples nls +png test webready +xmp"
```
`test` is yet here

geaaru commented on media-kit/mark/media-gfx/exiv2/templates/exiv2.tmpl:
```diff
@@ -0,0 +1,98 @@
+		-DEXIV2_ENABLE_WEBREADY=$(usex webready)
+		-DEXIV2_ENABLE_XMP=$(usex xmp)
+		-DEXIV2_ENABLE_BMFF=$(usex bmff)
+		$(echo -DEXIV2_BUILD_EXIV2_COMMAND=NO)
```
Is this something of prehistorical? It's not clear to me because we need an `echo`.

org-tekeli-borisp replied:
I have no idea

geaaru commented on media-kit/mark/media-gfx/exiv2/templates/exiv2.tmpl:
```diff
@@ -0,0 +1,98 @@
+		-DEXIV2_ENABLE_BMFF=$(usex bmff)
+		$(echo -DEXIV2_BUILD_EXIV2_COMMAND=NO)
+		$(echo -DEXIV2_BUILD_DOC=$(usex doc))
+		$(echo -DEXIV2_BUILD_UNIT_TESTS=$(usex test))
```
In addition of the already shared `echo` weird command to remove you need to drop `usex test` and force OFF. (i don't remember if OFF or NO is the correct value).

org-tekeli-borisp committed:
[media-kit] media-gfx/exiv2 Disable unit test build in exiv2 template.

This change prevents the inclusion of unit tests in the exiv2 build process by explicitly setting `DEXIV2_BUILD_UNIT_TESTS` to `NO`. It streamlines the build configuration and reduces unnecessary overhead.

org-tekeli-borisp commented:
Done!
![image](https://github.com/user-attachments/assets/a47958f5-3d5c-42d8-b012-0426521d91c4)

geaaru commented:
Last change my friend...
I think that we can drop all `echo`:
```
		$(echo -DEXIV2_BUILD_EXIV2_COMMAND=NO)
		$(echo -DEXIV2_BUILD_DOC=$(usex doc))
		$(echo -DEXIV2_BUILD_UNIT_TESTS=$(usex test))
```
in
```
		-DEXIV2_BUILD_EXIV2_COMMAND=NO
		-DEXIV2_BUILD_DOC=$(usex doc)
		-DEXIV2_BUILD_UNIT_TESTS=NO
```

org-tekeli-borisp committed:
Update exiv2.tmpl

drop echo

org-tekeli-borisp commented:
done!
